File: 2015/Day01/part2/solution.py
with open('input.txt') as f:
    data = f.read()

position = 0
count = 0
for char in data:
    position += 1
    count += 1 if char == '(' else -1 if char == ')' else 0
        
    if count == -1:
       print('Santa enters the basement at position', position)
       break

File: 2015/Day01/part2/solution.java
package Day01;

import java.util.Scanner;
import java.io.File;
import java.io.IOException;

public final class solution {
  
  public static void main(String[] args) throws IOException {

    Scanner sc = new Scanner(new File("input.txt"));
    String steps = sc.next();
    
    int count = 0;
    int position = 0;

    for (int i = 0; i < steps.length(); i++) {
      count += steps.charAt(i) == '(' ? 1 : -1;
      position++; 

      if (count == -1) {
        System.out.println("Santa enters the basement at position " + position);
        break;
      }
    }

    sc.close();
  }
}
